From 6798564b9e2f1a81b8c2d0cb4add97cb736d982b Mon Sep 17 00:00:00 2001
From: Mike Frysinger <vapier@gentoo.org>
Date: Mon, 16 Mar 2015 17:47:01 -0400
Subject: [PATCH] trylink: respect compiler settings when probing features

The CPPFLAGS/CFLAGS settings might have features that matter, so make
sure we utilize them when testing the compiler.

URL: https://bugs.gentoo.org/471118
Signed-off-by: Mike Frysinger <vapier@gentoo.org>
---
 scripts/trylink | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/scripts/trylink b/scripts/trylink
index 5a67fcf..48c487b 100755
--- a/scripts/trylink
+++ b/scripts/trylink
@@ -49,7 +49,7 @@ check_cc() {
     local tempname="$(mktemp)"
     # Can use "-o /dev/null", but older gcc tend to *unlink it* on failure! :(
     # "-xc": C language. "/dev/null" is an empty source file.
-    if $CC $1 -shared -xc /dev/null -o "$tempname".o >/dev/null 2>&1; then
+    if $CC $CPPFLAGS $CFLAGS $1 -shared -xc /dev/null -o "$tempname".o >/dev/null 2>&1; then
 	echo "$1";
     else
 	echo "$2";
@@ -66,7 +66,7 @@ check_libc_is_glibc() {
 	syntax error here
 	#endif
 	" >"$tempname".c
-    if $CC "$tempname".c -c -o "$tempname".o >/dev/null 2>&1; then
+    if $CC $CPPFLAGS $CFLAGS "$tempname".c -c -o "$tempname".o >/dev/null 2>&1; then
 	echo "$2";
     else
 	echo "$1";
-- 
2.3.2

